### ISSUE TITLE
Upgrade PyQt4 to PyQt5

### ISSUE THREAD

**Issue by CloudyPadmal** [fossasia/pslab-desktop#203]
There are many compatibility issues occuring due to the old version of PyQt libraries. They need to be upgraded to use the latest version to avoid issues as such.

**Comment by cweitat** [comment]
Would best if can take this opportunity to upgrade to Python 3 so that we won't have such issues.

**Comment by CloudyPadmal** [comment]
Yes @cweitat ! That's a good point. I'll take that also into account! :+1:

**Comment by cweitat** [comment]
@CloudyPadmal what's the progress of this? Did you face any problems?

**Comment by CloudyPadmal** [comment]
Hello @cweitat ! Yes, there were few issues as the app didn't get installed into python 3 and now I have implemented a method to get it installed into python 3. Then I had the PyQt4 version files not getting overwritten in pslab-desktop-apps repo. I tweaked some files and they seems processing. There are 50+ files need editing. I think I'm in the right track!

**Comment by CloudyPadmal** [comment]
Hi @cweitat ! I managed to get all the upgrades from python2 to python3 and PyQt4 to PyQt5. But there are method implementation from PyQt4 which are deprecated in PyQt5. As an example, `setMargins`. To support the new `setContentsMargin`, parameters need modifications and casting. So I'm currently stuck at this as there are quiet a lot of these methods need changes.

**Comment by cweitat** [comment]
@yatri1609 since you don't have the device, are you able to help Padmal on this?

**Comment by yatri1609** [comment]
Ya sure I'll help Padmal on this. Sorry for late reply.

**Comment by cweitat** [comment]
@CloudyPadmal is this finished?

**Comment by cweitat** [comment]
@CloudyPadmal able to finish by Thursday?

**Comment by CloudyPadmal** [comment]
I don't think I could. It includes redesigning of UI components, new object classes as the existing methods are depreciated with the newer version.

**Comment by fcartegnie** [comment]
Temporary patch to check and disable helpView
(untested)

```diff
diff --git a/PSL_Apps/bin/Experiments b/PSL_Apps/bin/Experiments
index 0bb6eb7..6385864 100755
--- a/PSL_Apps/bin/Experiments
+++ b/PSL_Apps/bin/Experiments
@@ -17,7 +17,15 @@ args = parser.parse_args()
 from PSL_Apps.utilitiesClass import utilitiesClass
 
 from PyQt4 import QtCore, QtGui
-from PyQt4 import QtWebKit
+
+# Deprecated webkit binding classes not available on Debian/Ubuntu
+global have_QtWebKit
+try:
+    from PyQt4 import QtWebKit
+    have_QtWebKit = 1
+except ImportError:
+    have_QtWebKit = 0
+    pass
 
 try:
     _fromUtf8 = QtCore.QString.fromUtf8
@@ -163,11 +171,14 @@ class MyMainWindow(QtGui.QMainWindow, single_col_exp.Ui_MainWindow,utilitiesClas
                self.menu_entries=[]
                self.menu_group=None
 
-               self.helpView = QtWebKit.QWebView()
-               self.helpLayout.addWidget(self.helpView)
-               self.helpView.setUrl(QtCore.QUrl("../../docs/MD_HTML/index.html"))
-               #from PSL_Apps.utilityApps.indexGen import autogen
-               #self.helpView.setHtml(autogen(pkg_resources.resource_listdir('psl_res','HTML')))
+                if have_QtWebKit:
+                    self.helpView = QtWebKit.QWebView()
+                    self.helpLayout.addWidget(self.helpView)
+                    self.helpView.setUrl(QtCore.QUrl("../../docs/MD_HTML/index.html"))
+                    #from PSL_Apps.utilityApps.indexGen import autogen
+                    #self.helpView.setHtml(autogen(pkg_resources.resource_listdir('psl_res','HTML')))
+                else:
+                    self.helpView = None
 
                self.shortlist=[]
                self.timer = QtCore.QTimer()
@@ -290,10 +301,11 @@ class MyMainWindow(QtGui.QMainWindow, single_col_exp.Ui_MainWindow,utilitiesClas
                        self.col=0;self.row+=1
 
        def setHelpUrl(self,fname): #fname must reside in psl_res.HTML
-               helpurl = pkg_resources.resource_filename('psl_res.HTML',fname)
-               if not os.path.isfile(helpurl): helpurl=None
-               self.helpView.setUrl(QtCore.QUrl(helpurl))          
-               self.tabWidget.setCurrentIndex(1)
+                if not self.helpView is None:
+                    helpurl = pkg_resources.resource_filename('psl_res.HTML',fname)
+                    if not os.path.isfile(helpurl): helpurl=None
+                    self.helpView.setUrl(QtCore.QUrl(helpurl))
+                    self.tabWidget.setCurrentIndex(1)
                
 
        def clearExperimentIcons(self):
@@ -323,8 +335,9 @@ class MyMainWindow(QtGui.QMainWindow, single_col_exp.Ui_MainWindow,utilitiesClas
 
                        self.runningApp = inst
                        #Load help HTML if found
-                       self.helpView.setUrl(QtCore.QUrl(helpurl))          
-                       self.tabWidget.setCurrentIndex(1)
+                        if not self.helpView is None:
+                           self.helpView.setUrl(QtCore.QUrl(helpurl))
+                           self.tabWidget.setCurrentIndex(1)
 
                else:
                        QtGui.QMessageBox.about(self,'Error','Hardware not detected')


```

**Comment by CloudyPadmal** [comment]
Fixed in #282
